crates/bevy_render/src/renderer/mod.rs

```diff
@@ -104,23 +104,70 @@ pub fn render_system(world: &mut World, state: &mut SystemState<Query<Entity, Wi
     }
 }
 
+/// A wrapper to safely make `wgpu` types Send / Sync on web with atomics enabled.
+/// On web with atomics enabled the inner value can only be accessed on the
+/// `wgpu` thread or else a panic will occur.
+/// On other platforms the wrapper simply contains the wrapped value.
+#[cfg(not(all(target_arch = "wasm32", target_feature = "atomics")))]
+#[derive(Debug, Clone)]
+pub struct WgpuWrapper<T>(T);
```

Comment:
It'd be ideal if this doesn't need to be public, as it's sort of an implementation detail. Perhaps we should make a wrapper for `Surface` since it's the only type used outside of bevy_render that needs this explicitly.

Reply:
OK sounds good to me. I believe @robtfm recently found that we may not need these wrappers anymore, but we can tackle that in a separate PR.